### PR TITLE
Feedstock for gRPC C++ libraries and command-line utilities

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@conda-forge/help-c-cpp The AppVeyor build succeeds here but for some reason the generated package is huge (each static library file is 100MB+), while it's quite small on Linux...

**Comment by pitrou** [comment]
@xhochy You might want to try this recipe on OS X since Travis-CI seems extremely slow.

File: recipes/grpc-cpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    (RPC) framework that can run anywhere. gRPC enables client and server
+    applications to communicate transparently, and simplifies the building
+    of connected systems.
+
+    This package contains the gRPC C++ libraries and header files, as well
+    as the code generation plugins.
+  doc_url: https://grpc.io/docs/
+
+extra:
+  recipe-maintainers:
```
Comment (xhochy) [inline recipes/grpc-cpp/meta.yaml]: Feel free to add me as a maintainer as well

**Comment by xhochy** [comment]
@pitrou could you also add the following diff:
```
diff --git a/recipes/grpc-cpp/build.sh b/recipes/grpc-cpp/build.sh
index 0c23cad88..059337f99 100755
--- a/recipes/grpc-cpp/build.sh
+++ b/recipes/grpc-cpp/build.sh
@@ -17,13 +17,16 @@ cmake ..  \
       -GNinja \
       -DCMAKE_BUILD_TYPE=Release \
       -DCMAKE_CXX_FLAGS="$CXXFLAGS" \
-      -DCMAKE_PREFIX_PATH=$CONDA_PREFIX \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
       -DCMAKE_INSTALL_PREFIX=$PREFIX \
       -DgRPC_CARES_PROVIDER="package" \
       -DgRPC_GFLAGS_PROVIDER="package" \
       -DgRPC_PROTOBUF_PROVIDER="package" \
       -DgRPC_SSL_PROVIDER="package" \
       -DgRPC_ZLIB_PROVIDER="package" \
+      -DCMAKE_AR=${AR} \
+      -DCMAKE_RANLIB=${RANLIB} \
+      -DProtobuf_ROOT=$PREFIX
```

This was required to get a local build running. Otherwise `cmake` picked up system defaults.

**Comment by pitrou** [comment]
Isn't PREFIX where the recipe is installed to, as opposed to the environment holding the dependencies?

**Comment by pitrou** [comment]
In other words, can you try with BUILD_PREFIX instead for `Protobuf_ROOT` and `CMAKE_PREFIX_PATH`?

**Comment by xhochy** [comment]
@pitrou I can give it a try. But locally I had `libprotobuf` in host as it should also be a library dependency?

**Comment by xhochy** [comment]
@pitrou no, that did not work.

**Comment by xhochy** [comment]
`PREFIX` worked for linking to `libprotobuf`.

**Comment by pitrou** [comment]
Ok, I'll switch to PREFIX then.

**Comment by xhochy** [comment]
@conda-forge/staged-recipes This is ready for merge.

**Comment by pitrou** [comment]
Ha, I didn't know we had to cc someone :-)

File: recipes/grpc-cpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  # Requires C++ 11, VS 2008 is not supported
+  skip: True  # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    # `protoc` is also used for building
+    - libprotobuf 3.6.1
```
Comment (scopatz) [inline recipes/grpc-cpp/meta.yaml]: Is libprotobuf really needed on the build system?  If you are linking to it, it should just be needed in host

Comment (xhochy) [reply recipes/grpc-cpp/meta.yaml]: `protoc` is required to compile protobuf definitions to sources.

File: recipes/grpc-cpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  host:
+    - c-ares
+    - gflags
+    - libprotobuf 3.6.1
+    - openssl
+    - zlib
+  run:
+    # Need to pin libprotobuf to the same version used when building,
+    # otherwise we may get a different SO ABI version.
+    - libprotobuf 3.6.1
```
Comment (isuruf) [inline recipes/grpc-cpp/meta.yaml]: This should be replaced by `{{ pin_compatible("libprotobuf", max_pin="x.x.x") }}`

File: recipes/grpc-cpp/meta.yaml (hunk 2 above)
Comment (isuruf) [inline recipes/grpc-cpp/meta.yaml]: This should only be needed in host

File: recipes/grpc-cpp/LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (isuruf) [inline recipes/grpc-cpp/LICENSE.txt]: Is this file in the tarball? Then you can remove it

File: recipes/grpc-cpp/meta.yaml (hunk 2 above)
Comment (wesm) [reply recipes/grpc-cpp/meta.yaml]: +1

File: recipes/grpc-cpp/LICENSE.txt (hunk 4 above)
Comment (pitrou) [reply recipes/grpc-cpp/LICENSE.txt]: Ah, I think I have misread the packaging instructions. Will do.

File: recipes/grpc-cpp/meta.yaml (hunk 3 above)
Comment (pitrou) [reply recipes/grpc-cpp/meta.yaml]: If I pass "x.x.x", will it ensure than the right bugfix version is depended upon at runtime (FTR, 3.6.0 is ABI-incompatible with 3.6.1...)?

Comment (isuruf) [reply recipes/grpc-cpp/meta.yaml]: Yes

**Comment by pitrou** [comment]
Hmm, the Azure Windows job failed with what looks like an unrelated error:
```
2019-02-06T15:17:55.6135081Z   File "C:\Miniconda\lib\site-packages\conda\core\subdir_data.py", line 60, in __call__
2019-02-06T15:17:55.6135425Z     assert channel.subdir
2019-02-06T15:17:55.6135782Z AssertionError
```

**Comment by pitrou** [comment]
Hmm, I had remove the wrong lisence file. Redoing it ;-)

**Comment by pitrou** [comment]
@conda-forge/staged-recipes Any other comments?

**Comment by scopatz** [comment]
LGTM other than the build number! Thanks @pitrou!

**Comment by pitrou** [comment]
Still a weird (unrelated?) error on Azure Pipelines.

**Comment by scopatz** [comment]
Yeah, you can ignore azure for the moment, and specifically windows builds.

**Comment by wesm** [comment]
@scopatz looks like this is ready to go, I'm anxious to start using this feedstock

**Pull request merged**